test/Dialect/FIRRTL/connect-errors.mlir

```diff
@@ -526,3 +526,25 @@ firrtl.module @test(in %a : !firrtl.uint<1>, out %b : !firrtl.sint<1>) {
   firrtl.connect %b, %a : !firrtl.sint<1>, !firrtl.uint<1>
 }
 }
+
+// -----
+
+/// Non-const types cannot be connected to const types
+
+firrtl.circuit "test" {
+firrtl.module @test(in %a : !firrtl.uint<1>, out %b : !firrtl.const.uint<1>) {
+  // expected-error @+1 {{type mismatch}}
+  firrtl.connect %b, %a : !firrtl.const.uint<1>, !firrtl.uint<1>
+}
+}
+
+// -----
+
+/// Const types cannot be connected to non-const types
+
+firrtl.circuit "test" {
+firrtl.module @test(in %a : !firrtl.const.uint<1>, out %b : !firrtl.uint<1>) {
+  // expected-error @+1 {{type mismatch}}
+  firrtl.connect %b, %a : !firrtl.uint<1>, !firrtl.const.uint<1>
```

Comment:
Note: this connection will likely be supported in a future pr

Reply:
Actually I just added support now since it's pretty minor and I'm pretty sure we won't be going with the `const-cast` approach